### ISSUE TITLE
Bug: [no-misused-promises ] should trigger on string concatenation

### ISSUE THREAD

**Issue by nartallax** [typescript-eslint/typescript-eslint#10381]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.6.3&fileType=.ts&code=MYewdgzgLgBAtgTwAoCcRwJYQKYwLwxjYDuMq6W2APNChmAOYB8AFCwJT5MwDeMAvuwBQoSLFHAAhlABK2CAFcANrAKJymHDADUMAEQR02AMpQ6jPUA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oDN4OBDfMwDmtYtA4BbSshTooiaBOiQANOGx4ipZBWp0GzNp1rTksWQBMxEs3IyRFyyBoC%2BIV0A&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
const myPromise = new Promise<string>(() => { })
const concatResult = myPromise + "someString"
```

### ESLint Config

```javascript
import eslint from '@eslint/js';
import tseslint from 'typescript-eslint';

let result = tseslint.config(
  eslint.configs.recommended,
  ...tseslint.configs.recommended,
  {
  files: ['**/*.{ts,tsx}'],
  languageOptions: {
    parserOptions: {
      project: "./tsconfig.json"
    },
    globals: {
      ...globals.browser,
    },
  },
  rules: {
    "@typescript-eslint/no-misused-promises": ["error", { "checksVoidReturn": false }]
  }
});

export default result
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

I expected that line 2 (in which promise is concatenated with string) would trigger no-misused-promises rule, because in most cases implicitly converting a promise to string is a mistake.

### Actual Result

There are no errors.

**Comment by bradzacher** [comment]
https://typescript-eslint.io/rules/no-base-to-string/
https://typescript-eslint.io/rules/restrict-plus-operands/
[example playground of above rules](https://typescript-eslint.io/play/#ts=5.6.3&fileType=.ts&code=MYewdgzgLgBAtgTwAoCcRwJYQKYwLwxjYDuMq6W2APNChmAOYB8AFCwJT5MwDeMAvuwBQoSLFHAAhlABK2CAFcANrAKJymHDADUMAEQR02AMpQ6jPUA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oDN4OBDfMwDmtYtA4BbSshTooiaBOiQANOGx4ipZBWp0GzNp1rTksWQBMxEs3IyRFytRpwES5KjXrJGLdlwARvyytPhcyPhUTMLyjkocKurYUO46et6G-tAoUZRkdMTwFrQcpND8TJaoDk6JkBoAviCNQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

Both rules are in our recommended set along side `no-misused-promises`.

I don't think there's much value in adding another check for the same issue.

**Comment by bradzacher** [comment]
This is why we recommend using recommended configs to enable all our most useful and most widely applicable error catching rules and you can build on top of the recommended config to enable/disable rules.
